tms/src/test/java/com/netcracker/edu/tms/dao/ProjectDaoJpaImplTest.java

```diff
@@ -0,0 +1,160 @@
+        database.getConnection().createStatement()
+                .execute(getContentByResourceRelativePath("deleteData.sql"));
+        database.getConnection().createStatement()
+                .execute(getContentByResourceRelativePath("fillData.sql"));
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        database.shutdown();
+    }
+
+    @Test
+    public void testGetProjectByIdExist() {
+        Project actual;
+        actual = projectDao.getProjectById(BigInteger.valueOf(1L));
+        Project expected = new Project(BigInteger.valueOf(1L), BigInteger.valueOf(1L), "name1");
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetProjectByIdNotExist() {
+        Project actual = projectDao.getProjectById(BigInteger.valueOf(10L));
+        Assert.assertNull(actual);
+    }
+
+    @Test
+    public void testGetProjectByCorrectNameWithProject() {
+        Project expected = new Project(BigInteger.ONE, BigInteger.ONE, "name1");
+        Project actual = projectDao.getProjectByName("name1");
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetProjectByCorrectNameWOProject() {
+        Project expected = null;
+        Project actual = projectDao.getProjectByName("name99");
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    @Transactional
+    public void testUpdateProjectWithExistentId() {
+        Project newProject = new Project(BigInteger.ONE, BigInteger.TEN, "name111");
+        Assert.assertTrue(projectDao.updateProject(newProject, newProject.getId()));
+    }
+
+    @Test
+    @Transactional
+    public void testUpdateProjectWithNonExistentId() {
+        Project newProject = new Project(BigInteger.TEN, BigInteger.TEN, "name111");
+        Assert.assertFalse(projectDao.updateProject(newProject, newProject.getId()));
+    }
+
+    @Test
+    @Transactional
+    public void testDeleteProjectWithExistentId() {
+        Assert.assertTrue(projectDao.deleteProject(BigInteger.ONE));
+    }
+
+    @Test
+    @Transactional
+    public void testDeleteProjectWithNonExistentId() {
+        Assert.assertFalse(projectDao.deleteProject(BigInteger.TEN));
+    }
+
+    @Test
+    public void testFindProjectsByCreatorIdWithProjects() {
+        List<Project> expected = new LinkedList<>();
+        expected.add(projectDao.getProjectById(BigInteger.ONE));
+        expected.add(projectDao.getProjectById(BigInteger.valueOf(6)));
+        List<Project> actual = projectDao.findProjectsByCreatorId(BigInteger.ONE);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testFindProjectsByCreatorIdWOProjects() {
+        List<Project> expected = null;
+        List<Project> actual = projectDao.findProjectsByCreatorId(BigInteger.TEN);
+        Assert.assertNotEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAllProjects() {
+        List<Project> expected = new LinkedList<>();
```

Comment:
What is the difference between ArrayList and LinkedList
Just leave it here https://twitter.com/joshbloch/status/583813919019573248 :)